src/Filesystem/Replicate.php

```diff
@@ -214,16 +216,52 @@ public function getAdapterClassNames()
 
     public function createFile($key, Filesystem $filesystem)
     {
-        return $this->primary->createFile($key, $filesystem);
+        if ($this->primary instanceof FileFactory) {
+            return $this->primary->createFile($key, $filesystem);
+        }
+
+        throw new \LogicException(sprintf('Primary adapter MUST implement %s.', FileFactory::class));
+    }
+
+    public function createStream(string $key, Filesystem $filesystem): Stream
+    {
+        if ($this->primary instanceof StreamFactory) {
+            return $this->primary->createStream($key, $filesystem);
+        }
+
+        throw new \LogicException(sprintf('Primary adapter MUST implement %s.', FileFactory::class));
```

Comment:
shouldn't you check for secondary too? if not, this change should go to master to when merging.

Reply:
As I'm not using this feature, I don't know if this change respects BC.
Are you aware why only the primary is currently used?

Reply:
I dont know either, thats why we changed it on master.

Reply:
Even if the argument doesn't sound very solid, in front of the lack of a better answer, I adhered the suggestion.